### PR TITLE
Add rhel9 support and new docker versions.

### PR DESCRIPTION
**Pull request by doflamingo721**
Following docker versions have been added as part of this PR:
````
20.10.24
23.0.3
23.0.4
23.0.5
23.0.6
````
The PR also adds docker installation support for RHEL 9 os by using CentOS packages as a workaround mentioned by upstream docker.

Linked issue:
https://github.com/rancher/rancher/issues/41192
https://github.com/rancher/rancher/issues/41454
https://github.com/rancher/rancher/issues/41526

### CODE REVIEW

**Review by snasovich — APPROVED** [review]
Compared `20.10.24.sh` and existing `20.10.23.sh` and the diff seems to properly cover allowing installations on RHEL (regardless of version though which is fine) with fallback to using CentOS repo for the package.

As being discussed offline, we can also reduce the number of versions we run automated tests so we don't hit 1h execution limit in Drone. Feel free to count the approval if it's the only thing that will be changed past this approval.

**Review by superseb — CHANGES_REQUESTED** [review]
I don't see a test for RHEL9, is that because there is none? Which images were tried for testing?

**Comment by doflamingo721** [comment]
@superseb Currently we don't have any tests for RHEL9 in the test-scripts. We are planning to add it but it will be tracked in a different ticket. I have tried installing these docker versions on RHEL9 and they were getting installed correctly.

**Review by snasovich — CHANGES_REQUESTED** [review]

**Comment by snasovich** [comment]
> @superseb Currently we don't have any tests for RHEL9 in the test-scripts. We are planning to add it but it will be tracked in a different ticket. I have tried installing these docker versions on RHEL9 and they were getting installed correctly.

@superseb @doflamingo721 , we've discussed this "offline" and there will be a separate ticket to add RHEL to automated tests. It's interesting that we didn't even have RHEL8 there already. :(

**Comment by superseb** [comment]
Why is the test separated? Isn't the point to test what we are changing? And looking up an image is really not much time to just handle this instead of postponing and then possibly forgetting.

@snasovich Version 8 is covered by both CentOS and Rocky, which is at least some coverage (and I believe it follows the same codepath).

File: scripts/test-docker [hunk 1]
```diff
@@ -13,14 +13,14 @@ while true; do
...
 
-test_docker_versions["ubuntu:18.04"]="^19.03.15$|^20.10|^23.0"
-test_docker_versions["ubuntu:20.04"]="^^19.03.15$|^20.10|^23.0"
-test_docker_versions["ubuntu:22.04"]="^20.10.1[3-9]|^20.10.2[0-9]|^23.0"
-test_docker_versions["centos:centos7"]="^19.03.15$|^20.10|^23.0"
-test_docker_versions["centos:centos8"]="^20.10.[7-9]|^20.10.1[0-9]|^20.10.2[0-9]|^23.0"
-test_docker_versions["oraclelinux:7"]="^19.03.15$|^20.10|^23.0"
-test_docker_versions["oraclelinux:8"]="^20.10|^23.0"
-test_docker_versions["rockylinux:8"]="^20.10.[8-9]|^20.10.1[0-9]|^20.10.2[0-9]|^23.0"
+test_docker_versions["ubuntu:18.04"]="^20.10.(20|[2-9]\d)|^23.0"
```
[the remark this answers is not in the document]
Comment (doflamingo721) [reply scripts/test-docker]: Corrected the regex. Can you please review it again?

**Comment by doflamingo721** [comment]
@snasovich @superseb I have added RHEL 9 to the automated testing script. Also created a separate [issue](https://github.com/rancher/rancher/issues/41526) to track this. I've linked the issue to this PR. Can you please review accordingly? Thanks in advance.

**Comment by doflamingo721** [comment]
@snasovich @superseb I have added RHEL 9 to the automated testing script. Also created a separate [issue](https://github.com/rancher/rancher/issues/41526) to track this. I've linked the issue to this PR. Can you please review accordingly? Thanks in advance.

File: scripts/test-docker [hunk 2]
```diff
@@ -13,14 +13,15 @@ while true; do
...
 
-test_docker_versions["ubuntu:18.04"]="^19.03.15$|^20.10|^23.0"
-test_docker_versions["ubuntu:20.04"]="^^19.03.15$|^20.10|^23.0"
-test_docker_versions["ubuntu:22.04"]="^20.10.1[3-9]|^20.10.2[0-9]|^23.0"
-test_docker_versions["centos:centos7"]="^19.03.15$|^20.10|^23.0"
-test_docker_versions["centos:centos8"]="^20.10.[7-9]|^20.10.1[0-9]|^20.10.2[0-9]|^23.0"
-test_docker_versions["oraclelinux:7"]="^19.03.15$|^20.10|^23.0"
-test_docker_versions["oraclelinux:8"]="^20.10|^23.0"
-test_docker_versions["rockylinux:8"]="^20.10.[8-9]|^20.10.1[0-9]|^20.10.2[0-9]|^23.0"
+test_docker_versions["ubuntu:18.04"]="^19.03.15$|^20.10.([2-9]\d)$|^23.0"
```
Comment (snasovich) [inline scripts/test-docker:16]: Nit, but do we actually need regex capturing group here (and in other occurrences below - except the one for `redhat/ubi9` where it's indeed needed)?
```suggestion
test_docker_versions["ubuntu:18.04"]="^19.03.15$|^20.10.[2-9]\d$|^23.0"
```

**Review by snasovich — APPROVED** [review]
One nit - otherwise LGTM. Thanks for adding RHEL9 to the tests!

File: scripts/test-docker (hunk 2 above)
Comment (snasovich) [reply scripts/test-docker:16]: Also just realized `.`s are unescaped and will match any character. But it's fine (and better for readability) - so we should leave those be.

Comment (doflamingo721) [reply scripts/test-docker]: That is correct, we don't need the capturing group. Updated the regex.

**Review by superseb — APPROVED** [review]

**Pull request merged**